pkg/assembler/backends/neo4j/isVulnerability.go

```diff
@@ -17,11 +17,187 @@ package neo4jBackend
 
 import (
 	"context"
-	"fmt"
+	"strings"
 
 	"github.com/guacsec/guac/pkg/assembler/graphql/model"
+	"github.com/neo4j/neo4j-go-driver/v4/neo4j"
+	"github.com/neo4j/neo4j-go-driver/v4/neo4j/dbtype"
+	"github.com/vektah/gqlparser/v2/gqlerror"
 )
 
 func (c *neo4jClient) IsVulnerability(ctx context.Context, isVulnerabilitySpec *model.IsVulnerabilitySpec) ([]*model.IsVulnerability, error) {
-	panic(fmt.Errorf("not implemented: IsVulnerability - IsVulnerability"))
+
+	queryAll := false
+	if isVulnerabilitySpec.Cve != nil && isVulnerabilitySpec.Ghsa != nil {
+		return nil, gqlerror.Errorf("cannot specify cve and ghsa together for IsVulnerability")
+	} else if isVulnerabilitySpec.Cve == nil && isVulnerabilitySpec.Ghsa == nil {
+		queryAll = true
+	}
+
+	session := c.driver.NewSession(neo4j.SessionConfig{AccessMode: neo4j.AccessModeRead})
+	defer session.Close()
+
+	aggregateIsVulnerability := []*model.IsVulnerability{}
+
+	if queryAll || isVulnerabilitySpec.Cve != nil {
+		var sb strings.Builder
+		var firstMatch bool = true
+		queryValues := map[string]any{}
+
+		// query CVE
+		query := "MATCH (root:Osv)-[:OsvHasID]->(osvID:OsvID)" +
+			"-[:subject]-(isVulnerability:IsVulnerability)-[:alias]-(cveID:CveID)<-[:CveHasID]" +
+			"-(cveYear:CveYear)<-[:CveIsYear]-(rootCve:Cve)"
```

Comment:
should edges to subject/objects be directed? else lgtm

Reply:
On query it's ok to not be directed if we don't have yet decided which direction to go. But for ingestion they have to be directed.

Reply:
Yes, this captures both directions but we can make it directed if needed in the future.